test/e2e/performanceprofile/functests/utils/node_inspector/inspector.go

```diff
@@ -19,58 +19,91 @@ import (
 	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
 	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/daemonset"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/images"
 	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/namespaces"
 	testpods "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/pods"
 )
 
 const serviceAccountSuffix = "sa"
 const clusterRoleSuffix = "cr"
 const clusterRoleBindingSuffix = "crb"
 
-func Create(cli client.Client, namespace, name, image string) error {
-	serviceAccountName := fmt.Sprintf("%s-%s", name, serviceAccountSuffix)
-	sa := createServiceAccount(serviceAccountName, namespace)
-	if err := cli.Create(context.Background(), sa); err != nil && !errors.IsAlreadyExists(err) {
+var initialized bool
+var namespace *corev1.Namespace = namespaces.NodeInspectorNamespace
+var nodeInspectorName = testutils.NodeInspectorName
+
+// initialize would be used to lazy initialize the node inspector
+func initialize() error {
+	if initialized {
+		return nil
+	}
+	// Create the test namespace
+	err := testclient.DataPlaneClient.Create(context.TODO(), namespace)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("Failed to create namespace: %v", err)
+	}
+
+	// Create Node Inspector resources
+	err = create()
+	if err != nil {
+		return fmt.Errorf("Failed to create Node Inspector resources: %v", err)
+	}
+
+	return nil
+}
+
+func create() error {
+	serviceAccountName := fmt.Sprintf("%s-%s", nodeInspectorName, serviceAccountSuffix)
+	sa := createServiceAccount(serviceAccountName, namespace.Name)
+	if err := testclient.DataPlaneClient.Create(context.Background(), sa); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
`errors.IsAlreadyExists(err)` deserves a separate handling.
Log a warning at least. 
And having the objects already create means the we didn't have a proper cleanup so it might had other sideaffects on the tests

Reply:
In case of `create()` this will be called once per each testing suite, because the initialized flag will be zeroed.
This follows the four create functions for the components, to verify they are indeed deployed. 
I feel that logging a warning implies unexpected behavior. but this is expected to happen, as the initialized flag is zerored.

Reply:
You should log a warning only if the components already exists.

If the components are already exist it means the cleanup of the previous run, was not handled as expected.

Reply:
> If the components are already exist it means the cleanup of the previous run, was not handled as expected.

This is not true in this case.
we don't cleanup in between suites. that means that for every suite, we are going to run this `create` function once and these component are expected to exist